core/src/main/javascript/crypto/KeyFormat.js

```diff
@@ -30,22 +33,27 @@
         /**
          * Normalize public key input into expected WebCrypto format.
          *
-         * @param {string|JSON|Uint8Array} input Base64-encoded or JSON or ByteArray of key.
-         * @param {string} format key format type ("spki" | "jwk")
-         * @return webcrypto format for the public key.
+         * @param {string|object|Uint8Array} input Base64-encoded or JSON object or ByteArray of key.
+         * @param {KeyFormat} format key format type ("spki" | "jwk")
+         * @return WebCrypto acceptable format of the public key.
          * @throws MslCryptoException if the key data is invalid.
          */
         normalizePubkeyInput: function normalizePubkeyInput(input, format) {
             if (format == KeyFormat.SPKI) {
                 try {
-                    input = (typeof input == "string") ? base64$decode(input) : input;
+                    input = (typeof input === "string") ? base64$decode(input) : input;
                 } catch (e) {
                     throw new MslCryptoException(MslError.INVALID_PUBLIC_KEY, format + " " + input, e);
                 }
             }
             else if (format == KeyFormat.JWK) {
                 try {
-                    input = (typeof input == "string") ? JSON.parse(input) : input;
+                    input = (typeof input === "string") ? JSON.parse(input) : input;
+                    /* input must be a JSON object */
+                    if (typeof input !== "object") {
```

Comment:
I've updated/tested/pushed a change that checks both of these conditions.

On Wed, Aug 3, 2016 at 1:23 PM, Wesley Miaw <notifications@github.com>
wrote:

> In core/src/main/javascript/crypto/KeyFormat.js
> <https://github.com/Netflix/msl/pull/126#discussion_r73413319>:
>
> >                  } catch (e) {
> >                      throw new MslCryptoException(MslError.INVALID_PUBLIC_KEY, format + " " + input, e);
> >                  }
> >              }
> >              else if (format == KeyFormat.JWK) {
> >                  try {
> > -                    input = (typeof input == "string") ? JSON.parse(input) : input;
> > +                    input = (typeof input === "string") ? JSON.parse(input) : input;
> > +                    /* input must be a JSON object */
> > +                    if (typeof input !== "object") {
>
> Actually you also need to do both checks: typeof x === "object" &&
> x.constructor === Object. Just typeof isn't good enough because it will
> also catch other things that are not pure JavaScript objects.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Netflix/msl/pull/126/files/1a843eda7f6b25ba0a3e0cddc74e828550afaef3..551ff692b66d2cce101c37884f043c37dcfb267f#r73413319>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATkwwxH9UidKF7MGvcZdHB-A-8YLlPy5ks5qcPi8gaJpZM4JYcJT>
> .
>